### ISSUE TITLE
Use the pretrained model retrain,problem occured！

### ISSUE THREAD

**Issue by NanhaiK** [davidsandberg/facenet#597]
hi,when i was trying to use the 20170512-110547 as pretrained model to retrain a new model,unfortunately,the problem "InvalidArgumentError (see above for traceback): Assign requires shapes of both tensors to match. lhs shape= [128,72] rhs shape= [128,44052] "occured.can you give me some help?many thanks.
environment: ubuntu16.04 ,python 3.6 ,tensorflow 1.2,cuda 8.0
btw,find some way try to solve the problem but did not work(such as delete the checkpoint)

**Comment by Heermosi** [comment]
Hello, are  you trying to train a smaller categories(72) on a trained model with larger categories(44052)?
If the output is a pure soft max, it's normal. Unless you replace the last classification layer(44052) with yours(72), and fix the weights of previous layers.

**Comment by NanhaiK** [comment]
thanks for your reply!command like this:
python train_softmax.py --logs_base_dir ./logs --models_base_dir v1 --data_dir hf06_mtcnnpy_182 --image_size 160 --model_def models.inception_resnet_v1 --lfw_dir lfw_mtcnnalign_160 --optimizer RMSPROP --learning_rate -1 --max_nrof_epochs 100 --keep_probability 0.8 --random_crop --random_flip --learning_rate_schedule_file learning_rate_schedule_classifier_casia.txt --weight_decay 5e-5 --center_loss_factor 1e-2 --center_loss_alfa 0.9 --epoch_size 90 --pretrained_model ./pretrain/20170512-110547/model-20170512-110547.ckpt-250000 --gpu_memory_fraction 0.9
I need to change the inception_resnet_v1? Could you give me some help?

**Comment by Heermosi** [comment]
The model before <logit softmax> was created in <network.inference>, the last part is created via slim.fully.connected.
Before I start, I must declare that I'm a green hand to this code, but I've spent months on tensorflow. So you should just take it as an advice.

1. You must load the model according to the training set, so that there would not be such error. The classification number is specified via <len(train_set)>. If you really understand what u r doing and you know exactly the number of categories, you may specify it via hard coding.

2. After you loaded the model from pretrained data, you must get rid of the last soft max layer. Because the last layer is used for train set classifications, the number of outputs must not be the same(together with the full connection weights) with what you want. **Make your own according to your train_set specified.**

3. Fix the weights before last layer, or just train it. The training usually would not change previous layers(as some scholars claimed).

4. Debug, understand how tensorflow works and debug it. If you do not have any experience before, I suggest you start with the MNIST examples put under tensorflow libs.

**Comment by Weijay** [comment]
Hi @Heermosi,
By doing so, it is kind of re-training the whole neural net (which takes a really long time) for my own data_set? chop off the classifier and re-train for my own classifier?

I just wonder how to add more classes to the pre-trained model. Suppose that the model originally classifies 10 categories, then I'd like to add another 10 categories, so my model can classify 20 categories. How can I do this?

**Comment by Linzai1994** [comment]
Alternatively,you can try run train_tripletloss.py by using pretrained model.I try it and everything looks normal.

**Comment by sarthakahuja11** [comment]
@Weijay  I have a similar problem. How did you solve it?
